superx/information_extractors/item_info_extractor.py

```diff
@@ -286,3 +322,28 @@ def get_num_of_pages(self):
                         break
 
         return num_of_pages
+
+    def create_branch_price_set(self):
+        """
+        queries the DB for all rows and saves the unique contraints in a set
+        """
+        all_rows = BranchPrice.query.all()
+
+        for row in all_rows:
+            unique_constraints = (row.chain_id,
+                                  row.item_code,
+                                  row.branch_id)
+            self.branch_price_unique_constraint_set.add(unique_constraints)
```

Comment:
This will be a crazy performance problem when you have thousands (or hundreds of thousands of rows). I suggest looking to see if there is a better way to do this.
For example, "Insert or ignore", "insert if not exist", "Insert or update"
There are ways to do all of these with sqlalchemy.

Another thing you might want to consider in the future is using asyncio or other asynchronous module for parsing multiple supermarkets in parallel

Reply:
I understand. The reasoning behind this is that before this method I queried every single obj anyways in order to see if it was in the db (see the prev commits). this causes the time of execution to be exponentially slower. By querying all and created a search dictionary I cause the bottleneck of the running time to be this method, which only runs 1 time. after that, all search functions extremely fast as I search the dictionary. 
In my opinion and from what I observed the run time for this will be <= querying for each object to see if it is in the db already.